diffkemp/simpll/Transforms.cpp

```diff
@@ -113,13 +113,33 @@ std::vector<FunPair> simplifyModulesDiff(Config &config) {
             mam.getResult<FunctionAbstractionsGenerator>(*config.Second,
                                                          config.SecondFun));
 
+    StringRef FirstFunName = config.FirstFun->getName();
+    StringRef SecondFunName = config.SecondFun->getName();
+
     // Module passes
     PassManager<Module, AnalysisManager<Module, Function *>, Function *,
         Module *> mpm;
     mpm.addPass(RemoveUnusedReturnValuesPass {});
     mpm.run(*config.First, mam, config.FirstFun, config.Second.get());
     mpm.run(*config.Second, mam, config.SecondFun, config.First.get());
 
+    // For the purpose of the comparision it is necessary to use the new
+    // function pointer in case the main function was replaced in order to
+    // compare the right functions.
+    if (config.FirstFun !=
+        config.First->getFunction(FirstFunName)) {
+        // Main function was replaced by a pass - fix the function pointer
+        config.FirstFun =
+                config.First->getFunction(FirstFunName);
+    }
+
+    if (config.SecondFun !=
+        config.Second->getFunction(SecondFunName)) {
+        // Main function was replaced by a pass - fix the function pointer
+        config.SecondFun =
+                config.Second->getFunction(SecondFunName);
+    }
+
```

Comment:
There is a method Config::refreshFunctions() available. It should do exactly what you need.